internal/anchors/check.go

```diff
@@ -0,0 +1,37 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package anchors
+
+import (
+	"errors"
+	"fmt"
+
+	"gopkg.in/yaml.v3"
+)
+
+func Check(n yaml.Node) error {
+	if n.Kind == yaml.AliasNode {
```

Comment:
I'm not super familiar with the use case, so I'm curious: Is there ever a case where we would want to allow one of these and not the other? Or is it always that we want to disable both at the same time? Sorry for not thinking of this on my first pass.

Reply:
(With the caveat that I'm not 100% sure of every corner of the YAML spec...) I don't think there's any legitimate use case for an alias node that isn't referenced by an anchor somewhere else. When you see them used, they're used together, where you define a reusable YAML chunk, and then reference it multiple places elsewhere.

This might mean there's a good lint pass to fail when there's a dangling unreferenced alias, or an anchor that points to an alias that doesn't exist (I may be using these terms wrong).

I'm personally more interested in avoiding the whole bee's nest and disallowing everything related to this, but I'm a curmudgeon 😄

Reply:
That sounds fine enough to me!